bot/exts/info/resources.py

```diff
@@ -0,0 +1,79 @@
+from typing import Optional
+
+import requests
+from bs4 import BeautifulSoup
+from discord import Embed
+from discord.ext import commands
+
+from bot.bot import Bot
+
+RESOURCES_URL = "https://www.pythondiscord.com/resources/"
+SITE_TOPICS_TITLE_CASE = []
+
+
+def get_topics() -> None:
+    """Scrape Python Discord website for updated list of topics to include in embed."""
+    topics_split = []
+    topics_title = ""
+    dont_capitalize = ["for", "and", "nor", "but", "or", "yet", "so", "in", "of", "a", "an"]
+    page = requests.get(RESOURCES_URL)
```

Comment:
Please see my other comments before spending too much time with this.

---

`requests` will be blocking the bot from doing other things. Take a look at [HTTPX](https://www.python-httpx.org/) which we use.

Create another function, that's defined as `async def` to be able to `await` when using `httpx`'s `AsyncClient`. Now, make the request and define a `content` variable set to the content of the request (similar to how you currently do `page.content`). Then, change `get_topics()` to take another parameter (which will be the content of the page) and remove the `requests.get(...)` call. Lastly, inside your new `async def` call it using `await asyncio.to_thread(get_topics, content)` (assuming `get_topics` is still the name of the function).

This will solve `requests` blocking the bot, as well as `bs4` potentially doing so as well.

Reply:
httpx is not used. aiohttp is.

Reply:
Ah sorry, yes that was false. [Both forms and pixels does though](https://cs.github.com/?scopeName=All+repos&scope=&q=org%3Apython-discord+httpx) which seems to have left a mark on me 😅.